### PR TITLE
Issue 36: User Login/Logout

### PR DESCRIPTION
**Pull request by pukkuKim**
## :: Type of change

- [X] New feature

## :: Description

User logging in/out via Google has been implemented.
**_Note_**: `verifyToken()` in `/server/src/middleware/auth.js` now works differently.

<br />

### CODE REVIEW

File: client/src/index.js [hunk 1]
```diff
@@ -9,10 +10,12 @@ import GlobalStyle from './global';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <React.StrictMode>
-    <Provider store={store}>
-      <GlobalStyle />
-      <App />
-    </Provider>
-  </React.StrictMode>
+  <GoogleOAuthProvider clientId="536529309790-aa4765f91fm6bebhp1jfhqhs9agj56b6.apps.googleusercontent.com">
```
Comment (jieunkxx) [inline client/src/index.js:13]: client id must be hidden. use .env instead pls

File: server/src/routes/user.js [hunk 2]
```diff
@@ -1,19 +1,70 @@
...
+const { FRONT_REDIRECT_URL } = process.env;
+
+const userRouter = Router();
 
 userRouter.get('/', verifyToken, asyncWrap(userController.getUser));
 userRouter.patch('/', verifyToken, asyncWrap(userController.updateUserInfo));
-userRouter.get('/', verifyToken, asyncWrap(userController.login));
 userRouter.post('/', asyncWrap(userController.signup));
 
+userRouter.get(
```
Comment (jieunkxx) [inline server/src/routes/user.js:20]: please hide the functions in controller/services/models

File: server/src/server.js [hunk 3]
```diff
@@ -2,25 +2,53 @@ import dotenv from 'dotenv';
...
+    saveUninitialized: false,
+    store: sessionStore,
+  })
+);
+app.use(passport.initialize());
+app.use(passport.authenticate('session'));
+passportIndex();
+
+const corsOptions = {
+  origin: 'http://localhost:3000',
```
Comment (jieunkxx) [inline server/src/server.js:47]: we need origin: '*'

**Review by jieunkxx — CHANGES_REQUESTED** [review]

File: server/src/server.js (hunk 3 above)
Comment (pukkuKim) [reply server/src/server.js:47]: Setting the `origin` as '*' did not work. May I request to investigate it together?

**Comment by pukkuKim** [comment]
`verifyToken` needs to be reviewed

**Pull request closed without merging**